src/test/java/com/ea/restaurant/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+package com.ea.restaurant.service.impl;
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import com.ea.restaurant.repository.UserRepository;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+class UserServiceImplTest {
+
+  @Mock UserRepository userRepository;
+
+  @InjectMocks UserServiceImpl userService;
```

Comment:
Use the interface instead of the implementation

Reply:
Done.